nextstrain_profiles/nextstrain-gisaid/config.yaml

```diff
@@ -9,4 +9,4 @@ show-failed-logs: True
 restart-times: 2
 reason: True
 stats: stats.json
-set-threads: tree=4
```

Comment:
This gave me a _very_ cryptic error when running locally:

```
Invalid threads definition: entries have to be defined as RULE=THREADS pairs (with THREADS being a positive integer).
Unparseable value: '{tree:'.
```
The error went away if I used a `shell` block within `make_auspice_config` rather than a `run` block. This may be an issue on my end, and commenting out this line isn't a good solution, but I ran out of debugging ideas!

Reply:
Hmm. It seems worth solving this before merge to me, or do you think not?

Can you paste more of the error/logs and how you're invoking the workflow? I can't reproduce this locally so far…

Reply:
I mean, I can reproduce the exact message if I do very specifically this, but obviously this is nonsense:

```diff
-# set-threads: tree=4
+set-threads: "{tree:"
```

Trying something less nonsense like:

```diff
-# set-threads: tree=4
+set-threads:
+  tree: 4
```

gets me a different (and reasonablish) error: `Unparseable value: "{'tree': 4}".`

Reply:
Yeah for sure I'd like to solve this - that's why I flagged it in review 😉 

```
$ snakemake --version
7.3.8

## with the "set-threads: tree=4" commented out
$ snakemake --cores 1 --profile nextstrain_profiles/nextstrain-gisaid/ -pf results/global/auspice_config.json
normal output and everything works as expected and creates the config JSON

## with "set-threads: tree=4"  in ./nextstrain_profiles/gisaid/config.yaml
## i.e. the config file is identical to current master
$ snakemake --cores 1 --profile nextstrain_profiles/nextstrain-gisaid/ -pf results/global/auspice_config.json
Config file defaults/parameters.yaml is extended by additional config specified via the command line.
Building DAG of jobs...
Using shell: /usr/local/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Conda environments: ignored
Job stats:
job               count    min threads    max threads
--------------  -------  -------------  -------------
auspice_config        1              1              1
total                 1              1              1

Cannot send slack message as the config does not define a channel and/or token.
Select jobs to execute...

[Thu May  5 13:07:21 2022]
Job 0: Making a custom auspice config.
Reason: Forced execution

Invalid threads definition: entries have to be defined as RULE=THREADS pairs (with THREADS being a positive integer). Unparseable value: '{tree:'. 
```

Reply:
I've reinstated the `set-threads` setting to [allow an AWS test run](https://github.com/nextstrain/ncov/actions/runs/2273245463). If that passes then I'm happy to go with it, as that'll be n=2 setups which can't reproduce the issue, but I would really like to know what's going on!

Reply:
Ah, this is a Snakemake version issue. The Conda environment defined by this repo gets a 6.x version of Snakemake and the Nextstrain Docker runtime has a 5.x version, neither of which exhibit the "Invalid threads definition" error. However, if I use 7.3.8, 7.4.0, or 7.5.0 then I can reproduce. The error goes away again with 7.6.0.

Looking at the code/issues finds these:

- https://github.com/snakemake/snakemake/commit/0771062a07f0e2cfe9ee45a2276aa61b096eb6e1
- https://github.com/snakemake/snakemake/pull/1617
- https://github.com/snakemake/snakemake/issues/1622

Reply:
Ouch. Well, at least we've figured it out.

> The Conda environment defined by this repo gets a 6.x version of Snakemake

I'm missing something... where is this environment? We have `workflow/envs/nextstrain.yaml`, but that doesn't specify snakemake (as it's intended to be run by snakemake itself). The [ncov docs](https://docs.nextstrain.org/projects/ncov/en/latest/tutorial/example-data.html) say to install via the [nextstrain docs](https://docs.nextstrain.org/en/latest/install.html) which don't pin versions at all: `mamba create ... snakemake`. Augur itself specifies `"snakemake >=5.4.0"`.

Reply:
Ah, you're right. I was confused. `workflow/envs/nextstrain.yaml` does include Snakemake transitively and it gets 6.8.0 (not the latest), but the `mamba create` command in our install docs gets latest Snakemake (7.6.1).

Reply:
(There was something else going on yesterday where I tried created a new Conda env with just Snakemake and got an older version, but I now suspect that related to missing channels and/or whatever the default Python pin was.)